gapis/api/vulkan/api/copy_clear_commands.api

```diff
@@ -676,19 +676,23 @@ sub void copyImageBuffer(VkBuffer buffer, VkImage image, VkImageLayout layout, m
           yInExtent := y - yStart
           rowStartByte := depthStartByte + y * imageLayout.rowPitch
           rowStartBlockInExtent := ((zInExtent * imageHeight) + yInExtent) * rowLength
+          bufStart := as!VkDeviceSize(bufferLayerOffset + rowStartBlockInExtent * elementSizeInBuffer)
+          bufSize := as!VkDeviceSize((xEnd - xStart) * elementSizeInBuffer)
+          readMemoryInBuffer(bufferObject, bufStart, bufSize)
+          bufMemPieces := getBufferBoundMemoryPiecesInRange(bufferObject, bufStart, bufSize)
+          imgStart := rowStartByte + xStart * elementSizeInImage
           if isPackedDepth {
-            for x in (xStart .. xEnd) {
-              imgStart := as!VkDeviceSize(rowStartByte + x * elementSizeInImage)
-              rowStartInExtent := (rowStartBlockInExtent + x) * elementSizeInBuffer
-              bufStart := as!VkDeviceSize(bufferLayerOffset + rowStartInExtent)
-              readMemoryInBuffer(bufferObject, bufStart, as!VkDeviceSize(elementSizeInBuffer))
-              // Discard the 4th byte in the buffer for current pixel.
-              bufMemPieces := getBufferBoundMemoryPiecesInRange(bufferObject, bufStart, as!VkDeviceSize(elementSizeInImage))
-              for _ , _ , piece in bufMemPieces {
-                bufMemStart := piece.MemoryOffset
-                bufMemEnd := bufMemStart + piece.Size
-                imgPieceStart := (piece.ResourceOffset - bufStart) + imgStart
-                imgPieceEnd := imgPieceStart + piece.Size
+            for _ , _ , piece in bufMemPieces {
+              // The below assumes that each memory piece is alligned to a multiple of
+              // elementSizeInBuffer. This has to be true, since the smallest alignment of pages is
+              // 64 bytes, and elementSizeInBuffer is 4.
```

Comment:
if ```elementSizeInBuffer is 4``` why do we have this stored in a variable?

Reply:
because it can have different values in different parts of the code.

Reply:
also, and sorry if that's what you originally meant to ask about, not using `4` here, even though we know it's `4` is because `elementSizeInBuffer` is more descriptive (esp next to `elementSizeInImage`) than `4`.

Reply:
Would you care about adding an assert there to make sure it's _always_ 4?

Reply:
We don't have an assert mechanism in API file, as far as I know.